### ISSUE TITLE
Asciicast file format specs -- Precision about timestamps

### ISSUE THREAD

**Issue by s-leroux** [asciinema/asciinema#386]
I looked at the [v2 specs](https://github.com/asciinema/asciinema/blob/develop/doc/asciicast-v2.md#event-stream) but I didn't see anything saying that events MUST be ordered according to their timestamp.

May I assume in a _valid_ asciicast file, events are stored in a _strictly increasing timestamps order_, or is it valid:

* to have out-of-order events,
* to have several consecutive events with the _same_ timestamp?

<hr>

FWIW, the Python player has no issue with multiple identical timestamps (neither with _decreasing_ timestamps, by the way) since it will process events in the file's order anyway. But for more advanced tasks (like #385) allowing multiple events with the same timestamp makes things like precisely indexing an event complicated.

**Comment by sickill** [comment]
Good catch. All of the projects that read asciicast files, at least the ones under asciinema GH org, assume strictly monotonic timestamps. This is especially important for reading/processing asciicasts in a streaming fashion. So I think we should just update v2 spec doc to be explicit about the necessity for monotonic timestamps / chronological events.
